src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,73 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React from 'react';
+import { iconWrapperStyle, wrapperStyle } from './Chip.style';
+import { colorShades, flatColors } from 'theme/palette';
+import { flexCenterVertical } from 'theme/functions';
+import { TestId } from 'utils/types';
+import { generateTestDataId } from 'utils/helpers';
+
+export type Props = {
+  /**
+   * Style of input field
+   * @default filled
+   */
+  styleType?: 'filled' | 'outlined';
+  /** This property define the size of the button.
+   *  @default md
+   * */
+  size?: 'md' | 'sm';
+  /** Defines the fill color of the component, if filled */
+  fill?: typeof flatColors[number];
+  /** Defines the shade of the fill */
+  shade?: typeof colorShades[number];
+  /** An optional icon to show to the left */
+  leftIcon?: JSX.Element;
+  /** */
+  onLeftIconClick?: React.ReactEventHandler;
+  /** An optional icon to show to the right */
+  rightIcon?: JSX.Element;
+  /** */
+  onRightIconClick?: React.ReactEventHandler;
+  /**  */
+};
+
+type TestProps = {
+  dataTestId?: TestId;
+};
+
+const Chip: React.FC<Props & TestProps> = ({
+  styleType = 'filled',
+  size = 'md',
+  fill,
+  shade,
+  leftIcon,
+  onLeftIconClick: leftIconHandler,
+  rightIcon,
+  onRightIconClick: rightIconHandler,
+  dataTestId = '',
+  children,
+}) => {
+  return (
+    <div css={[flexCenterVertical]}>
```

Comment:
@panvourtsis I am a bit concern with this outer div. It was in my initial implementation because of how TextField has an outer div, but even if I believe that here is unnecessary, problems occur when I remove it (and I havent found a solution yet). As shown in the 2nd screenshot, extra width is added to the chip (but only) when icons are used and I cannot pinpoint to the cause
![Screenshot 2020-12-10 at 8 45 58 AM](https://user-images.githubusercontent.com/18506343/101732643-37db8100-3ac6-11eb-8834-99555cea2f1e.png)
![Screenshot 2020-12-10 at 8 59 22 AM](https://user-images.githubusercontent.com/18506343/101732653-3ca03500-3ac6-11eb-9c5d-d0e6e65a246e.png)